eo-parser/src/main/java/org/eolang/parser/StrictXmir.java

```diff
@@ -0,0 +1,181 @@
+import java.util.List;
+import javax.xml.namespace.NamespaceContext;
+import org.cactoos.bytes.BytesOf;
+import org.cactoos.bytes.IoCheckedBytes;
+import org.cactoos.io.InputOf;
+import org.w3c.dom.Node;
+import org.xembly.Directives;
+import org.xembly.Xembler;
+import org.xml.sax.SAXParseException;
+
+/**
+ * XMIR that validates itself right after construction.
+ *
+ * <p>This class is supposed to be used ONLY for testing, because
+ * it modifies the XML encapsulated: it replaces the location of
+ * XSD schema with a file, thus making testing much faster.</p>
+ *
+ * @since 0.49.0
+ */
+@SuppressWarnings("PMD.TooManyMethods")
+public final class StrictXmir implements XML {
+
+    /**
+     * The XML.
+     */
+    private final XML xml;
+
+    /**
+     * Ctor.
+     * @param src The source
+     */
+    public StrictXmir(final XML src) {
+        this.xml = new StrictXML(StrictXmir.reset(src));
+    }
+
+    @Override
+    public String toString() {
+        return this.xml.toString();
+    }
+
+    @Override
+    public List<String> xpath(final String query) {
+        return this.xml.xpath(query);
+    }
+
+    @Override
+    public List<XML> nodes(final String query) {
+        return this.xml.nodes(query);
+    }
+
+    @Override
+    public XML registerNs(final String prefix, final Object uri) {
+        return this.xml.registerNs(prefix, uri);
+    }
+
+    @Override
+    public XML merge(final NamespaceContext context) {
+        return this.xml.merge(context);
+    }
+
+    @Override
+    @Deprecated
+    public Node node() {
+        throw new UnsupportedOperationException("deprecated");
+    }
+
+    @Override
+    public Node inner() {
+        return this.xml.inner();
+    }
+
+    @Override
+    public Node deepCopy() {
+        return this.xml.deepCopy();
+    }
+
+    @Override
+    public Collection<SAXParseException> validate() {
+        return this.xml.validate();
+    }
+
+    @Override
+    public Collection<SAXParseException> validate(final XML schema) {
+        return this.xml.validate(schema);
+    }
+
+    /**
+     * Here, we check the location of the XSD in the XML
+     * and replace with a new one, if necessary.
+     * @param xml Original XML
+     * @return New XML with the same node
+     */
+    private static XML reset(final XML xml) {
+        final Node node = xml.inner();
+        final List<String> location = xml.xpath("/program/@xsi:noNamespaceSchemaLocation");
+        if (!location.isEmpty()) {
+            String uri = location.get(0);
+            if (uri.startsWith("http")) {
+                uri = String.format(
+                    "file:///%s",
+                    StrictXmir.download(
+                        uri,
+                        Paths.get("target/xsd").resolve(
+                            uri.substring(uri.lastIndexOf('/') + 1)
+                        )
+                    ).toString().replace("\\", "/")
+                );
+            }
+            new Xembler(
```

Comment:
@yegor256 Do we need to change the `xmir` in case `!uri.startWith("http")`?

Reply:
@volodya-lombrozo nope, when it's `file://....` we keep it as is

Reply:
@yegor256 I see. I've overlooked indentations. Thank you.